File: 0621-task-scheduler/0621-task-scheduler.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        if(n==0)
            return tasks.size();
        vector<int>freq(26,0);
        for(auto ch:tasks)freq[ch-'A']++;
        sort(freq.begin(),freq.end(),greater<int>());
        int cntr=0;
        for(int i=0;i<26;i++)cntr+=(freq[0]==freq[i]);
        return max((freq[0]-1)*(n+1)+cntr,(int)tasks.size());
    }
};

File: 0621-task-scheduler/0621-task-scheduler.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] freq=new int[26];
        int mx=0,cnt=0;
        for(char task:tasks)
        {
            freq[task-'A']++;
            mx=Math.max(mx,freq[task-'A']);
        }
        for(int f:freq)cnt+=(f==mx)?1:0;
        int space = (mx-1)*(n);
        int left = tasks.length-mx-space-cnt+1;
        
        int res = space + mx + (cnt-1);
        if(left<=0)
        {
            return res;
        }
        else
        {
            return res+left;
        }
    }
}